thenewboston_node/business_logic/storages/path_optimized_file_system.py

```diff
@@ -50,8 +50,9 @@ def list_directory(self, directory_path, sort_direction=1):
 
     def _list_directory_generator(self, directory_path):
         for dir_path, _, filenames in os.walk(directory_path):
+            filenames = map(strip_compression_extension, filenames)
+            filenames = set(filenames)  # remove duplicated files after strip
```

Comment:
let's also log a warning if duplicate files are detected (no need for filenames, just the fact is enough)